### ISSUE TITLE
[inference] load_lora_onto_transformer discards the load result, so an unapplied adapter silently renders the base

### ISSUE THREAD

**Issue by aramintak** [alvdansen/signet-trainer#105]
`inference/lora_load.py::load_lora_onto_transformer` (168-198) ends with:

```python
wrapped = get_peft_model(transformer, lora_config)
set_peft_model_state_dict(wrapped.get_base_model(), state_dict)
return wrapped
```

The return value is discarded. `set_peft_model_state_dict` does not raise on a key mismatch — it returns `_IncompatibleKeys` and applies whatever lined up. A `lora_B` left at its zero init is an exact identity, so an adapter that lands nowhere renders **pixel-identical to the base model under an adapter label**, with nothing in the output indicating it.

This is the same failure class that `lora/peft.py::load_adapter_state_into` was hardened against, where it is described as *"the paid-run failure the audit named"* and closed with three raises (empty file / unexpected keys / cold keys). That hardening is not shared with this loader.

`load_lora_onto_transformer` is live: `modal/fns.py::sample()` calls it at 7 sites (1887, 1984, 2092, 2261, 2655, 2916).

**Ask:** route this loader through the hardened `load_adapter_state_into`, or apply the same three checks here.
